test/azlinux_test.go

```diff
@@ -1685,3 +1694,83 @@ func testLinuxSymlinkArtifacts(ctx context.Context, t *testing.T, cfg testLinuxC
 		assert.NilError(t, err)
 	})
 }
+
+func testImageConfig(ctx context.Context, t *testing.T, target string, opts ...srOpt) {
+	spec := &dalec.Spec{
+		Name:        "test-image-config",
+		Version:     "0.0.1",
+		Revision:    "42",
+		Description: "Test to make sure image configs are copied over",
+		License:     "MIT",
+		Image: &dalec.ImageConfig{
+			Entrypoint: "some-entrypoint",
+			Cmd:        "some-cmd",
+			Env: []string{
+				"ENV1=VAL1",
+				"ENV2=VAL2",
+			},
+			Labels: map[string]string{
+				"label.1": "value1",
+				"label.2": "value2",
+			},
+			Volumes: map[string]struct{}{
+				"/some/volume": {},
+			},
+			WorkingDir: "/some/work/dir",
+			StopSignal: "SOME-SIG",
+			User:       "some-user",
+		},
+	}
+
+	envToMap := func(envs []string) map[string]string {
+		out := make(map[string]string, len(envs))
+		for _, env := range envs {
+			k, v, _ := strings.Cut(env, "=")
+			out[k] = v
+		}
+		return out
+	}
+
+	testEnv.RunTest(ctx, t, func(ctx context.Context, gwc gwclient.Client) {
+		opts = append(opts, withSpec(ctx, t, spec))
+		opts = append(opts, withBuildTarget(target))
+		sr := newSolveRequest(opts...)
+		res := solveT(ctx, t, gwc, sr)
+
+		dt, ok := res.Metadata[exptypes.ExporterImageConfigKey]
+		assert.Assert(t, ok, "missing image config in result metadata")
+
+		var img dalec.DockerImageSpec
+		err := json.Unmarshal(dt, &img)
+		assert.NilError(t, err)
+
+		assert.Check(t, cmp.Equal(strings.Join(img.Config.Entrypoint, " "), spec.Image.Entrypoint))
+		assert.Check(t, cmp.Equal(strings.Join(img.Config.Cmd, " "), spec.Image.Cmd))
+
+		// Envs are merged together with the base image
+		// So we need to validate that the values we've set are what we expect
+		// Often there will at least one other env for `PATH` we won't check
```

Comment:
No sure exactly what this means but is there a typo too?
```suggestion
		// Often there will be at least one other env for `PATH` we won't check
```